src/merkle/path_set.rs

```diff
@@ -16,15 +16,30 @@ impl MerklePathSet {
     // --------------------------------------------------------------------------------------------
 
     /// Returns an empty MerklePathSet.
-    pub fn new(depth: u8) -> Result<Self, MerkleError> {
+    pub fn new(depth: u8) -> Self {
         let root = [ZERO; 4];
         let paths = BTreeMap::new();
 
-        Ok(Self {
+        Self {
             root,
             total_depth: depth,
             paths,
-        })
+        }
+    }
+
+    /// Appends the provided paths iterator into the set.
+    ///
+    /// Analogous to `[Self::add_path]`.
+    pub fn with_paths<I>(self, paths: I) -> Result<Self, MerkleError>
+    where
+        I: IntoIterator<Item = (u64, Word, MerklePath)>,
+    {
+        paths
+            .into_iter()
+            .try_fold(self, |mut set, (index, value, path)| {
+                set.add_path(index, value, path)?;
+                Ok(set)
+            })
```

Comment:
@Al-Kindi-0 - this address the comment I left [here](https://github.com/0xPolygonMiden/miden-vm/pull/705#discussion_r1111325948). The interface is slightly different - but it should accomplish the same thing.

Reply:
Will make the switch once it is merged.